docs_src/security/SeedingServiceSecrets.md

```diff
@@ -0,0 +1,88 @@
+# Seeding Service Secrets
+
+!!! edgey "EdgeX 2.1"
+    New for EdgeX 2.1 is the ability to seed service specific secrets during the service's start-up. 
+
+All EdgeX services now have the capability to specify a JSON file that contains the service's secrets which are seeded into the service's `SecretStore` during service start-up. This allows the secrets to be present in the service's `SecretStore` when the service needs to use them.
+
+The new `SecretsFile` setting on the `SecretStore` configuration allows the service to specify the fully-qualified path to the location of the service's secrets file. Normally this setting is left blank when a service has no secrets to be seeded.
+
+!!! example "Example setting SecretsFilePath in TOML"
+
+    ```toml
+    [SecretStore]
+    Type = "vault"
+    ...
+    SecretsFile = "/tmp/my-service/secrets.json"
+    ...
+    ```
+
+This setting can also be overriding with the `SECRETSTORE_SECRETSFILE` environment variable in the docker-compose file and then volume mounted in to the service's container.
+
+!!! example "Example setting SecretsFilePath via environment override"
+    ```yaml
+    environment:
+      SECRETSTORE_SECRETSFILEPATH: "/tmp/my-service/secrets.json"
+      ...
+    volumes:
+    - /tmp/my-service/secrets.json:/tmp/my-service/secrets.json
+    ```
+
+During service start-up, after `SecretStore` initialization, the service's secrets JSON file is read, validated and the secrets stored into the service's `SecretStore`. The file is then rewrite without the sensitive secret data that was successfully stored. 

```

Comment:
**suggest**: read, validated, and the secrets...

Reply:
_The file is then rewrite without_ --> the file is then **rewritten** without...

Reply:
fixed both

Reply:
I don't think we should require the file to be writable.  If it is, we should scrub the secret out.  Otherwise skip doing so.  Or make it a configurable option.

Reply:
> I don't think we should require the file to be writable. If it is, we should scrub the secret out. Otherwise skip doing so. Or make it a configurable option.

@bnevis-i , so what I hear is if can write the file, do so w/o the secret data (as it is implemented now). If the file is not writeable, skip and log a warning? This is directly related to the comment on the code PR here https://github.com/edgexfoundry/go-mod-bootstrap/pull/276#discussion_r716781049

Reply:
https://github.com/edgexfoundry/go-mod-bootstrap/pull/276#discussion_r716907161